12-offer-encoding.md

```diff
@@ -551,144 +582,100 @@ using the `onion_message` `invoice` field.
     - SHOULD ignore any payment which does not use one of the paths.
-  - MUST specify `amount`.`msat` in multiples of the minimum lightning-payable unit
-    (e.g. milli-satoshis for bitcoin) for `chain` (or for bitcoin, if there is no `chain`).
   - if responding to an `invoice_request`:
-    - if for the same `offer_id` and `payer_key` as a previous `invoice_request`:
+    - if `invoice_request_payer_key` and offer are identical to a previous `invoice_request`:
       - MAY simply reuse the previous invoice.
     - otherwise:
       - MUST NOT reuse a previous invoice.
-    - MUST set `node_id` the same as the offer.
-    - MUST set (or not set) `quantity` exactly as the invoice_request did.
-    - MUST set `payer_key` exactly as the invoice_request did.
-    - MUST set (or not set) `payer_info` exactly as the invoice_request did.
-    - MUST set (or not set) `payer_note` exactly as the invoice_request did,
-      or MUST not set it.
-    - MUST set (or not set) `replace_invoice` exactly as the invoice_request did.
-    - MUST begin `description` with the `description` from the offer.
-    - MAY append additional information to `description` (e.g. " +shipping").
-    - MUST set `amount` to the *base invoice amount* calculated from the invoice_request
-    - MUST set (or not set) `issuer` exactly as the offer did.
-    - MUST NOT set `refund_for`
-    - MUST NOT set `refund_signature`
-  - otherwise (responding to a `send_invoice` offer):
-    - MUST set `node_id` to the id of the node to send payment to.
-    - MUST set `description` the same as the offer.
-    - if the offer had a `quantity_min` or `quantity_max` field:
-      - MUST set `quantity`
+    - if `invoice_request_amount` is not present:
+      - MUST set `invoice_request_amount`.`msat` to the *base invoice amount* calculated from the invoice_request
+  - otherwise (responding to a `offer_send_invoice` offer):
+    - MUST fail the request if `offer_send_invoice` is not present.
+    - MUST fail the request if the offer fields not do exactly match a valid, unexpired offer.
+    - if `offer_quantity_min` or `offer_quantity_max` are present:
+      - MUST set `invoice_request_quantity`
       - MUST set it within that (inclusive) range.
     - otherwise:
-      - MUST NOT set `quantity`
-    - MUST set `payer_key` to the `node_id` of the offer.
-    - MUST NOT set `payer_info`.
-    - MUST set (or not set) `refund_for` exactly as the offer did.
-    - if it sets `refund_for`:
-      - MUST set `refund_signature` to the signature of the
-        `refunded_payment_hash` using prefix `refund_signature` and the `payer_key` from the to-be-refunded invoice.
-    - otherwise:
-      - MUST NOT set `refund_signature`
+      - MUST NOT set `invoice_request_quantity`
+    - MUST set `invoice_request_payer_key` to `offer_node_id`.
+    - MUST set `invoice_request_payer_info`
+	  - SHOULD set it to at least 16 random bytes.
```

Comment:
There's no invoice_request in this case (above it says:   - otherwise (responding to a `offer_send_invoice` offer):).

From a high level, an 'send_invoice' offer *is* an invoice request.  Now invoice_request *has all the offer fields*, do we just, y'know, use it directly instead of having this weird kind of offer?  I need to think through this...